pkg/endpoint/endpoint.go

```diff
@@ -301,8 +302,7 @@ type Endpoint struct {
 	// realizedRedirects maps the ID of each proxy redirect that has been
 	// successfully added into a proxy for this endpoint, to the redirect's
 	// proxy port number.
-	// You must hold Endpoint.mutex to read or write it.
-	realizedRedirects map[string]uint16
+	realizedRedirects sync.Map
```

Comment:
Changing this code will be a tiny bit less readable/safe without the typed map, I wonder if we should have a thin typed generic wrapper around this type.

Reply:
That would certainly be useful in this case. I wonder if there's one in the standard-ish library (/x, /pkg) already.

Reply:
I've decided to revert this part of the change.